File: Ahorn/entities/RuinsSentinel.jl
module RuinsSentinel

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/RuinsSentinel" RuinsSentinel(x::Integer, y::Integer, attackRange::Integer=60, detectionRange::Integer=150, health::Integer=3, moveSpeed::Integer=50, patrolDistance::Integer=100)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(RuinsSentinel)
)

function Ahorn.selection(entity::RuinsSentinel)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::RuinsSentinel, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: Loenn/entities/ruinsSentinel.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local ruinsSentinel = {}

ruinsSentinel.name = "MaggyHelper/RuinsSentinel"
ruinsSentinel.depth = -100
ruinsSentinel.fieldInformation = {
    health = {
        minimumValue = 1
    },
    detectionRange = {
        minimumValue = 0
    },
    attackRange = {
        minimumValue = 0
    },
    moveSpeed = {
        minimumValue = 0
    },
    patrolDistance = {
        minimumValue = 0
    }
}
ruinsSentinel.placements = {
    {
        name = "normal",
        data = {
            health = 3,
            detectionRange = 150,
            attackRange = 60,
            moveSpeed = 50,
            patrolDistance = 100
        }
    }
}

function ruinsSentinel.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/ruins_sentinel/dormant", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function ruinsSentinel.selection(room, entity)
    return utils.rectangle(entity.x - 12, entity.y - 28, 24, 28)
end

return ruinsSentinel
